### PR TITLE
Fixes 5052: refactor jobs and temp/repo association

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

This pulls 2 refactors out of https://github.com/content-services/content-sources-backend/pull/865
* refactors jobs into a single command  ./cmd/jobs/main.go
 * removes old unused jobs
* refactors the template_repository_configurations association into an official model

## Testing steps

1.  Tests pass
2.  running `go run cmd/jobs/main.go retry_failed_task`  seems to run the appropriate code

### CODE REVIEW

File: deployments/deployment.yaml [hunk 1]
```diff
@@ -323,74 +323,16 @@ objects:
               - mountPath: /tmp
                 name: tmpdir
       jobs:
-        - name: retry-failed-tasks
+        - name: rename-domains
```
Comment (xbhouse) [inline deployments/deployment.yaml:326]: this and line 335 might have been missed in the refactor. i think it should be `retry_failed_tasks`?

Comment (jlsherrill) [reply deployments/deployment.yaml:326]: good catch!

Comment (jlsherrill) [reply deployments/deployment.yaml]: fixed!

File: deployments/deployment.yaml [hunk 2]
```diff
@@ -332,7 +332,7 @@ objects:
             inheritEnv: true
             command:
               - /jobs
-              - rename_domains
+              - retry-failed-tasks
```
Comment (xbhouse) [inline deployments/deployment.yaml:335]: oh sorry i think the arg here should be `retry_failed_task` (singular with underscores)

```
bhouse@mac:content-sources-backend $ go run cmd/jobs/main.go retry_failed_task
9:09AM WRN Updated 0 tasks
```
```
bhouse@mac:content-sources-backend $ go run cmd/jobs/main.go retry-failed-tasks
9:09AM WRN Usage: go run cmd/jobs/main.go  $JOB_NAME
  (Possible jobs: retry_failed_task)
exit status 255
```

Comment (jlsherrill) [reply deployments/deployment.yaml:335]: doh

Comment (jlsherrill) [reply deployments/deployment.yaml]: fixed

**Pull request merged**